packages/common/src/components/CondaHint.ts

```diff
@@ -0,0 +1,300 @@
+import CodeMirror, { Editor } from 'codemirror';
+import 'codemirror/addon/hint/show-hint';
+import 'codemirror/addon/hint/show-hint.css';
+
+function getChannels(environment_yml: string) {
+  const matches = [
+    ...environment_yml.matchAll(/^channels:\s*\n((?:\s+- \S+\s*\n)+)\S/gm)
+  ];
+  if (!matches[0] || matches[0].length < 2) {
+    return;
+  }
+  const channels = matches[0][1];
+  if (!channels) {
+    return;
+  }
+  return channels
+    .split('\n')
+    .filter(x => x)
+    .map(line => line.match(/\s+- (.+)\s*/)[1]);
+}
+
+function getDependencies(environment_yml: string) {
+  const matches = [
+    ...environment_yml.matchAll(/^dependencies:\s*\n((?:\s+- .*(\n|$))+)/gm)
+  ];
+  if (!matches[0] || matches[0].length < 2) {
+    return;
+  }
+  const dependencies = matches[0][1];
+  if (!dependencies) {
+    return;
+  }
+  return dependencies
+    .split('\n')
+    .filter(x => x)
+    .map(line => line.match(/\s+- (.+)/)[1]);
+}
+
+export function getName(environment_yml: string): string | null {
+  const matches = [...environment_yml.matchAll(/^name:\s*(.*?)\s*(?:\n|$)/gm)];
+  if (!matches[0] || matches[0].length < 2) {
+    return;
+  }
+  return matches[0][1];
+}
+
+async function loadChannels(quetzUrl: string) {
+  const response = await fetch(`${quetzUrl}/api/channels`);
```

Comment:
Should the request not be proxied through the server extension to avoid CORS error?

Reply:
I think the public quetz server will have CORS headers to allow this. The benefit of doing this directly in the front-end is less latency and less code. The downside is having to get `subdir` from the backend and having no way to query private channels.